src/usecases/disassociate-design-use-case.ts

```diff
@@ -49,16 +55,24 @@ export const disassociateDesignUseCase = {
 				entity.id,
 			);
 
-			const [design, issue] = await Promise.all([
-				figmaService.getDesign(figmaDesignId, {
-					atlassianUserId,
-					connectInstallationId: connectInstallation.id,
-				}),
-				jiraService.getIssue(disassociateFrom.id, connectInstallation),
-			]);
+			// By setting updateSequenceNumber to 0 the design will not be updated,
+			// and therefore we don't need to fetch the actual design data.
+			const designStub = {
```

Comment:
**suggestion (non-blocking):** We could expose the desired API from `JiraService` and  encapsulate the implementation there.

For example:

```ts
export const disassociateDesignUseCase = {
   // ...
   
   // Under the hood, `JiraService.deleteDesignAssociation` creates the stub and use
   // available Jira API to implement the desired behavior. Later, this implementation can be
   // replaced with no effect on the use case.
   await jiraService.deleteDesignAssociation(entity.id, designIssueAssociation);
   
    // ...
}
```